sympy/physics/mechanics/tests/test_functions.py

```diff
@@ -250,3 +252,76 @@ def test_center_of_mass():
     point_o.set_pos(p1.point, center_of_mass(p1.point, p1, p2, p3, p4, b))
     expr = 5/(m + mb + 6)*a.x + (m + mb + 3)/(m + mb + 6)*a.y + mb/(m + mb + 6)*a.z
     assert point_o.pos_from(p1.point)-expr == 0
+
+def test_apply_torque():
+    t = symbols('t')
+    q = dynamicsymbols('q')
+    B1 = Body('B1')
+    B2 = Body('B2')
+    N = ReferenceFrame('N')
+    torque = t*q*N.x
+
+    apply_torque(torque, B1, B2) #Applying equal and opposite torque
+    assert B1.loads == [(B1.frame, torque)]
+    assert B2.loads == [(B2.frame, -torque)]
+
+    torque2 = t*N.y
+    apply_torque(torque2, B1)
+    assert B1.loads == [(B1.frame, torque+torque2)]
+
+def test_apply_force():
+    f, g = symbols('f g')
+    q, x, v1, v2 = dynamicsymbols('q x v1 v2')
+    P1 = Point('P1')
+    P2 = Point('P2')
+    B1 = Body('B1')
+    B2 = Body('B2')
+    N = ReferenceFrame('N')
+
+    P1.set_vel(B1.frame, v1*B1.x)
+    P2.set_vel(B2.frame, v2*B2.x)
+    force = f*q*N.z # time varying force
+
+    apply_force(force, B1, B2, P1, P2) #applying equal and opposite force on moving points
+    assert B1.loads == [(P1, force)]
+    assert B2.loads == [(P2, -force)]
+
+    g1 = B1.mass*g*N.y
+    g2 = B2.mass*g*N.y
+
+    apply_force(g1, B1) #applying gravity on B1 masscenter
+    apply_force(g2, B2) #applying gravity on B2 masscenter
+
+    assert B1.loads == [(P1,force), (B1.masscenter, g1)]
+    assert B2.loads == [(P2, -force), (B2.masscenter, g2)]
+
+    force2 = x*N.x
+
+    apply_force(force2, B1, B2) #Applying time varying force on masscenter
+
+    assert B1.loads == [(P1, force), (B1.masscenter, force2+g1)]
+    assert B2.loads == [(P2, -force), (B2.masscenter, -force2+g2)]
+
+def test_apply_loads_on_multi_degree_freedom_holonomic_system():
+    W = Body('W') #Wall
+    B = Body('B') #Block
+    P = Body('P') #Pendulum
+    b = Body('b') #bob
+    q1, q2 = dynamicsymbols('q1 q2') #generalized coordinates
+    k, c, g, kT = symbols('k c g kT') #constants
+    F, T = dynamicsymbols('F T') #Specified forces
+
+    #Applying forces
+    B.apply_force(F*W.x)
+    apply_force(k*q1*W.x, W, B) #Spring force
+    apply_force(c*q1.diff()*W.x, W, B) #dampner
+    P.apply_force(P.mass*g*W.y)
+    b.apply_force(b.mass*g*W.y)
+
+    #Applying torques
+    apply_torque(kT*q3*W.z, P, b)
+    P.apply_torque(T*W.z)
```

Comment:
This should be equal and opposite between B and P.

Reply:
I think `kT*q3*W.z` is equal and opposite. `T*W.z` is only on P.

Reply:
It is between B and P. With no assumption of a newtonian frame, all forces/torques should be account for. Also this is a unit test, it is helpful to show that the general cases work, not just special cases.